stdlib/map.ml

```diff
@@ -356,4 +367,63 @@ module Make(Ord: OrderedType) = struct
 
     let choose = min_binding
 
+    let add_list m l = List.fold_left (fun acc (k,v) -> add k v acc) m l
+
+    let of_list l = add_list empty l
+
+    let to_list m = fold (fun k v acc -> (k,v) :: acc) m []
+
+    let rec add_gen m g = match g() with
+      | None -> m
+      | Some (k,v) -> add_gen (add k v m) g
+
+    let of_gen g = add_gen empty g
+
+    let to_gen m =
```

Comment:
I don't think this iterates in ascending order. (Do your tests check this?)

See [Batteries implementation](https://github.com/ocaml-batteries-team/batteries-included/blob/88c10f3/src/batMap.ml#L325) with the `iter` type to represent iteration state, the `cons_iter` function that conses a tree to (the left of) an iterator, and the `enum_next` function for generation from an iterator.